packages/amplify-e2e-tests/src/rds-v2-tests-common/rds-auth-userpool-fields.ts

```diff
@@ -0,0 +1,1925 @@
+      expect(onUpdateSubscriptionResult).toHaveLength(1);
+      expect(onUpdateSubscriptionResult[0].data[`onUpdate${modelName}`].id).toEqual(todoRandom.id);
+      expectNullFields(onUpdateSubscriptionResult[0].data[`onUpdate${modelName}`], ['owner', 'authors', 'privateContent', 'publicContent', 'ownerContent', 'ownersContent']);
+      */
+      const onDeleteSubscriptionResult = await subTodoHelper.subscribe('onDelete', []);
+      expect(onDeleteSubscriptionResult).toHaveLength(0);
+    });
+
+    test('Custom owner model auth and allowed field operations', async () => {
+      const modelName = 'TodoCustomOwnerContentVarious';
+      const user1ModelOperationHelpers = createModelOperationHelpers(appSyncClients[userPoolProvider][userName1], schema);
+      const user2ModelOperationHelpers = createModelOperationHelpers(appSyncClients[userPoolProvider][userName2], schema);
+      const user1TodoHelper = user1ModelOperationHelpers[modelName];
+      const user2TodoHelper = user2ModelOperationHelpers[modelName];
+
+      const todo = {
+        customId: Date.now().toString(),
+        author: userName1,
+        privateContent: 'Private Content',
+        ownerContent: 'Owner Content',
+      };
+      const createResultSetName = `create${modelName}`;
+      const updateResultSetName = `update${modelName}`;
+      const deleteResultSetName = `delete${modelName}`;
+      const privateResultSet = `
+        customId
+        privateContent
+      `;
+      const ownerResultSet = `
+        ${privateResultSet}
+        author
+      `;
+      const completeOwnerResultSet = `
+        ${ownerResultSet}
+        ownerContent
+      `;
+
+      // owner(user1) creates a record with only allowed fields
+      const createResult1 = await user1TodoHelper.create(createResultSetName, todo, completeOwnerResultSet);
+      checkOperationResult(createResult1, { ...todo, privateContent: null, ownerContent: null }, createResultSetName);
+
+      // user1 can update the allowed fields
+      const todoUpdated1 = {
+        customId: todo['customId'],
+        author: userName1,
+        privateContent: 'Private Content updated',
+      };
+      const updateResult1 = await user1TodoHelper.update(updateResultSetName, todoUpdated1, completeOwnerResultSet);
+      checkOperationResult(updateResult1, { ...todo, ...todoUpdated1, privateContent: null, ownerContent: null }, updateResultSetName);
+
+      // user1 can read the allowed fields
+      const getResult1 = await user1TodoHelper.get(
+        {
+          customId: todo['customId'],
+        },
+        completeOwnerResultSet,
+        false,
+        'all',
+        'customId',
+      );
+      checkOperationResult(getResult1, { ...todo, ...todoUpdated1 }, `get${modelName}`);
+
+      const listTodosResult1 = await user1TodoHelper.list({}, completeOwnerResultSet, `list${modelName}`, false);
+      checkListItemExistence(listTodosResult1, `list${modelName}`, todo['customId'], true, 'customId');
+
+      /*
```

Comment:
As above, add an easily searchable todo, or link to this code in our backlog ticket